cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -628,13 +634,17 @@ private List<ApplicationId> decodeAndValidateBatchApplication(NamespaceId namesp
           throw new BadRequestException("Missing 'appId' in the request element.");
         }
         String appId = obj.get("appId").getAsString();
-
-        JsonElement version = obj.get("version");
-        if (version == null) {
-          result.add(validateApplicationId(namespaceId, appId));
+        String version;
+        JsonElement versionObj = obj.get("version");
+        if (versionObj == null || versionObj.equals("-SNAPSHOT")) {
+          // get the latest version of the app instead of -SNAPSHOT
+            ApplicationDetail appLatestVersionDetail = applicationLifecycleService.getLatestAppDetail
+              (new ApplicationId(namespaceId.getNamespace(), appId));
```

Comment:
This is a dangerous operation as we are hitting the DB many times inside a for-loop. Please pull this out to a list of app name that we need to fetch the latest version and use the multiGet underneath to fetch the result in one query to the DB.

Reply:
This means we need a new table method that does `multiScan(Collection<Range> keyRanges, int limit, Field<?> filterIndex)`, current table cannot do a multiScan on the latest field

Reply:
can we do this:  use our existing scan with namespace and filter latest == true (retrieve the set of latest apps) first and then in the for loop compare if this app-name in the json-object is present ? This is loading all the apps in memory tho - how bad is that?

Reply:
@chtyim @sanjanasandeep 
This is doable if we relax the restriction further on the given range -- let it accept non-primary keys. Another gain from this is we can solve the UI history versions pagination issue -- we can scan on namespace+application+creation_time

I can push the table update if we want, or use @sanjanasandeep suggestion and add a TODO task?